src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClientCertificateManager.cs

```diff
@@ -176,10 +176,13 @@ public static void InstallLocalCertificateFromBridge()
                 }
             }
 
-            InstallCertificateToMyStore(certificateToInstall);
-
-            // We also need to install the root cert if we install a local cert
-            InstallRootCertificateFromBridge();
+            // certificateToInstall could be null in the case the user certification exists
```

Comment:
certificateToInstall is null when foundUserCertificate is true, as the isLocalKeyName is not set, thus it continues to  InstallCertificateToMyStore.

The bug here might be why isLocalKeyName is not set from server. I'll open a issue to investigate it. Don't want to block this checkin.